app/gui2/src/util/assert.ts

```diff
@@ -6,6 +6,10 @@ export function assert(condition: boolean): asserts condition {
   if (!condition) bail('Assertion failed')
 }
 
+export function assertFn(pred: () => boolean) {
+  if (!pred()) bail('Assertion failed: ' + pred.toString().replace(/^\s*\([^)]*\)\s*=>\s*/, ''))
+}
```

Comment:
This assert variant misses the most important feature, which is the typescript's `asserts condition` guard. I'm not sure if there is a way to do the same with a closure return value, but probably not. As for printing the expression, we do have a callstack for the standard assertions, which I believe is enough to identify what have failed. So, is this function ever useful?

Reply:
probably not, the intent was for it to be more convenient to declare assertions with decent DX when they fail, without having to explicitly specify an error message. i don't believe it's currently used though so it should probably be removed for now anyway.